### PR TITLE
Fix: Form Grid strings are translatable and account for singular or plural usages

### LINKED ISSUE #6587: All donation form grid strings should be translatable
## Details

Some strings on the donation form grid are not being properly translated. There also also WP functions that could be refactored in a more accurate readable way.

## Expected Behavior

All strings are translated on the donation form grid.

## Acceptance Criteria

-   [x] All strings on the donation form grid  are translatable.
-   [x] Singular or Plural text for the goal amounts are properly displayed.

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves #6587

## Description

This fixes and adds translations on several strings. The WP function _n() which translates a string to either a singular or plural version based on a number value was not working correctly. The original container element that was used in the _n() ,  has been moved outside the conditional statement and translations have been simplified by being inserted directly into the html. A Translation has also been added to a "Donation" string that was previously not translated.

## Affects
FormGrid

## Visuals

https://user-images.githubusercontent.com/75056371/196825716-6d36a84b-1748-4a79-9208-625fce1981db.mov

## Testing Instructions

- Create a form and set the donation goal type to either donors or donations.
- Set the goal amount to 1.
- Make a donation.
- Create a page with the FormGrid shortcode or block.
- View the the bottom left & verify "1 of 1 donor or donation"
- Edit the form goal to 25.
- View the bottom left of the Form Grid & verify "1 of 25 donors or donations"

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203211562108141

### CODE REVIEW

File: templates/shortcode-form-grid.php [hunk 1]
```diff
@@ -74,388 +74,371 @@ static function($term)use($tag_text_color,$tag_bg_color){
...
+                }
 
-                    // Maybe display the form title.
-                    if ( true === $atts['show_title'] ) {
-                        printf(
-                            '<h3 class="give-form-grid-content__title">%1$s</h3>',
-                            $formTemplate->getFormHeading( $form_id )
+                // Maybe display the form excerpt.
+                if ( true === $atts['show_excerpt'] ) {
+                    if ( $raw_content = $formTemplate->getFormExcerpt( $form_id ) ) {
```
Comment (kjohnson) [inline templates/shortcode-form-grid.php:163]: Is this a related change? Or rather, is this intended to be a change in this PR?

File: templates/shortcode-form-grid.php [hunk 2]
```diff
@@ -364,10 +365,8 @@ static function($term)use($tag_text_color,$tag_bg_color){
                                             );
                                             echo sprintf(
                                             /* translators: 1: amount of income raised 2: goal target amount. */
-                                                __( '<div class="form-grid-raised__details">
-                                                   <span class="amount"  data-amounts="%1$s">%2$s</span>
-                                                   <span class="goal" data-amounts="%3$s"><span>of </span>%4$s</span>
-                                         </div>', 'give' ),
+                                                __( '<span class="amount"  data-amounts="%1$s">%2$s</span>
+                                                     <span class="goal" data-amounts="%3$s"><span>of </span>%4$s</span>', 'give' ),
```
Comment (kjohnson) [inline templates/shortcode-form-grid.php]: Should these even be translatable? Goal target amount is a number, no?

File: templates/shortcode-form-grid.php [hunk 3]
```diff
@@ -377,25 +376,22 @@ static function($term)use($tag_text_color,$tag_bg_color){
...
                                                            <span class="amount">%1$s</span>
-                                                           <span class="goal"><span>of</span> %2$s <span>Donation</span></span>
-                                                </div>',
-                                                    '<div class="form-grid-raised__details">
+                                                           <span class="goal"><span>of</span> %2$s <span>Donation</span></span>',
+                                                    '
                                                            <span class="amount">%1$s</span>
-                                                           <span class="goal"><span>of</span> %2$s <span>Donations</span></span>
-                                                </div>',
+                                                           <span class="goal"><span>of</span> %2$s <span>Donations</span></span>',
```
Comment (kjohnson) [inline templates/shortcode-form-grid.php]: I would think that only `Donation/Donations` should be translated here. The surrounding HTML just complicates the translation. Also, I'm sure that we have `Donation/Donations` translated elsewhere in the plugin, so we could re-use that value if we isolate what needs to be translated.

File: templates/shortcode-form-grid.php [hunk 4]
```diff
@@ -378,41 +378,40 @@ static function($term)use($tag_text_color,$tag_bg_color){
...
 
                                         <div class="form-grid-raised__details">
                                             <?php
-                                            $amountText = _n('donation', 'donations',  $form->get_sales(), 'give');
 
-                                            echo sprintf(
-                                                '<span class="amount form-grid-raised__details_donations">%s</span>
-                                                        <span class="goal">%s</span>', $form->get_sales(), $amountText ) ?>
+                                            echo sprintf( /* translators: 1: total number of donations completed */
+                                                __('<span class="amount form-grid-raised__details_donations">%1$s</span>
```
Comment (kjohnson) [inline templates/shortcode-form-grid.php]: This HTML partial doesn't seem to need to be translatable. The `__n()` call below translates the only translatable text in the string.

File: templates/shortcode-form-grid.php (hunk 1 above)
Comment (JoshuaHungDinh) [reply templates/shortcode-form-grid.php:163]: @kjohnson No changes were made it was a formatting difference.
 Heres a photo of the files current state you can see both functions remain in the same order Screen Shot 2022-10-20 at 9 43 03 AM

File: templates/shortcode-form-grid.php (hunk 4 above)
Comment (JoshuaHungDinh) [reply templates/shortcode-form-grid.php]: @kjohnson I ended up refactoring the sprintfs inside the tags to make it easier to read in https://github.com/impress-org/givewp/pull/6589/commits/1664d81f0cf0a12b9def3dc3fd65d4f1f6a60c35

Screen Shot 2022-10-20 at 9 56 32 AM

File: templates/shortcode-form-grid.php (hunk 2 above)
Comment (JoshuaHungDinh) [reply templates/shortcode-form-grid.php]: @kjohnson Yea it's a number.  The word "of" is in the span. I also remove the surrounding span in a later commit

File: templates/shortcode-form-grid.php (hunk 3 above)
Comment (JoshuaHungDinh) [reply templates/shortcode-form-grid.php]: @kjohnson I ended up refactoring the sprintfs inside the tags to make it easier to read in https://github.com/impress-org/givewp/pull/6589/commits/1664d81f0cf0a12b9def3dc3fd65d4f1f6a60c35

Screen Shot 2022-10-20 at 9 56 32 AM

File: templates/shortcode-form-grid.php (hunk 2 above)
Comment (kjohnson) [reply templates/shortcode-form-grid.php]: Ah, I missed that part. Ok.

File: templates/shortcode-form-grid.php (hunk 3 above)
Comment (kjohnson) [reply templates/shortcode-form-grid.php]: Perfect.

**Review by kjohnson — APPROVED** [review]
@JoshuaHungDinh thanks for addressing my concerns. Let's get this sent off to QA. Could you add this change to the QA pipeline in Asana?

**Review by rickalday — APPROVED** [review]
Passed manual tests
